### PR TITLE
cabana: add keyboard shortcuts to binary view

### PR DESCRIPTION
**Pull request by pd0wm**
- Delete signal: x, backspace, delete
- Toggle endianness: e
- Toggle signedness: s
- Open chart: c, p, g (Chart, Plot, Graph)

Any thoughts on how to explain this to the user?

### CODE REVIEW

**Comment by pd0wm** [comment]
@deanlee any ideas on how to cleanly set the edited/new signal to be the new `hovered_sig`?

**Comment by deanlee** [comment]
there is a signals in SIgnalView:

> signals:
  void highlight(const Signal *sig);

`emit highlight(sig)` can notify bin view to set `hovered_sig` to `sig` . Is this what you need?

**Comment by deanlee** [comment]
call BinaryView::highlight(signal) or `emit highlight(sig)` in SignalView. Do these two methods meet your needs?

**Comment by pd0wm** [comment]
How do I get a pointer to the new signal?

**Comment by deanlee** [comment]
the old pointer stay valid.

File: tools/cabana/binaryview.cc [hunk 1]
```diff
@@ -62,19 +62,27 @@ void BinaryView::addShortcuts() {
   QShortcut *shortcut_endian = new QShortcut(QKeySequence(Qt::Key_E), this);
   QObject::connect(shortcut_endian, &QShortcut::activated, [=]{
     if (hovered_sig != nullptr) {
+      const Signal *hovered_sig_prev = hovered_sig;
       Signal s = *hovered_sig;
       s.is_little_endian = !s.is_little_endian;
       emit editSignal(hovered_sig, s);
+
+      hovered_sig = nullptr;
+      highlight(hovered_sig_prev);
```
Comment (deanlee) [inline tools/cabana/binaryview.cc:71]: can call highlight(hovered_sig) here, hovered_sig will not changed after refresh, no need to save it to a temp variable hovered_sig_prev.

**Comment by deanlee** [comment]
> Any thoughts on how to explain this to the user?

added an online-help: https://github.com/commaai/openpilot/pull/27349 .

**Pull request merged**